IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m

```diff
@@ -463,6 +463,17 @@ -(void)notifyFinishedNavigation:(NSURL *)url webView:(WKWebView *)webView
     [self stopSpinner];
 }
 
+- (NSString *)getRefreshTokenCredential {
+    
+    if (_customHeaders != nil && _customHeaders.count > 0)
```

Comment:
Would it make sense to make this functionality a bit more generic? It seems incorrect to me that MSIDOAuth2EmbeddedWebviewController has knowledge about one specific custom header that is being used for a specific scenario. Would it perhaps make more sense to pass along all other headers that were provided by the caller? Or maybe have caller specify which headers need to be passed along? Additionally, any AAD specific logic might be better to be included inside the MSIDAADOAuthEmbeddedWebviewController file instead of a generic MSIDOAuth2EmbeddedWebviewController file.

Reply:
good point and I follow the reasoning. I originally implemented it that way but thought not to expose all _customHeaders as it is private to the OAuth2WVController and not expose them and enable them to be mutable. I can make them readyonly. Based on the nature of being generic handler, you are right it should not be knowledgeable of specific cases. i will fix it.